pyo3-macros-backend/src/pyclass.rs

```diff
@@ -515,6 +520,52 @@ fn impl_class(
     let is_basetype = attr.is_basetype;
     let is_subclass = attr.has_extends;
 
+    // If the pyclass has extends/unsendable, we must opt back into PyCell checking
+    // so that the inner Rust object is not inappropriately shared between threads.
```

Comment:
I'm not sure whether this is sufficient.

I didn't see another obvious way around `unsendable` other than to re-enable borrow tracking.